design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,321 @@
+}
+
+interface GiftRegistrySearcheableAttributeMetadataInterface {
+    is_searcheable: Boolean!
+}
+
+interface GiftRegistrySelectAttributeMetadataInterface {
+    options: [GiftRegistrySelectAttributeOptionMetadata]
+}
+
+type GiftRegistrySelectAttributeOptionMetadata {
+    code: ID!
+    label: String!
+    is_default: Boolean
+}
+
+type GiftRegistry {
+    id: ID!
+    event_name: String!
+    type: GiftRegistryType
+    message: String!
+    created_on: String! @doc(description: "Creation date") @doc(description: "Accessible to the registry owner only")
+    privacy_settings: GiftRegistryPrivacySettings! @doc(description: "Accessible to the registry owner only")
+    status: GiftRegistryStatus! @doc(description: "Accessible to the registry owner only")
+    owner_name: String!
+    registrants: [GiftRegistryRegistrant]
+    shipping_address: CustomerAddress @doc(description: "Accessible to the registry owner only")
+    dynamic_attributes: [GiftRegistryDynamicAttribute]
+    items: [GiftRegistryItemInterface]
+}
+
+interface GiftRegistryItemInterface {
+    id: String!
+    quantity: Float!
+    quantity_fulfilled: Float!
+    note: String
+    added_on: String!
+    product: ProductInterface
+    selected_customizable_options: [SelectedCustomizableOption]
+}
+
+type SimpleGiftRegistryItem implements GiftRegistryItemInterface {
+}
+
+type BundleGiftRegistryItem implements GiftRegistryItemInterface {
+    selected_bundle_options: [SelectedBundleOption!]
+}
+
+type ConfigurableGiftRegistryItem implements GiftRegistryItemInterface {
+    selected_configurable_options: [SelectedConfigurableOption!]
+}
+
+# Not currently supported by Magento core
+type DownloadableGiftRegistryItem implements GiftRegistryItemInterface {
+    links: [DownloadableProductLinks]
+    samples: [DownloadableProductSamples]
+}
+
+# Not currently supported by Magento core
+type VirtualGiftRegistryItem implements GiftRegistryItemInterface {
+}
+
+# Not currently supported by Magento core
+type GiftCardGiftRegistryItem implements GiftRegistryItemInterface {
+    sender_name: String!
+    recepient_name: String!
+    amount: Money!
+    message: String
+}
+
+interface GiftRegistryDynamicAttributeInterface  {
+    code: ID!
+    label: String!
+    value: String!
+}
+
+type GiftRegistryDynamicAttribute implements GiftRegistryDynamicAttributeInterface {
+    group: GiftRegistryDynamicAttributeGroup!
+}
+
+type GiftRegistryRegistrantDynamicAttribute implements GiftRegistryDynamicAttributeInterface {
+
+}
+
+enum GiftRegistryDynamicAttributeGroup {
+    GENERAL_INFORMATION
+    EVENT_INFORMATION
+    PRIVACY_SETTINGS
+    DETAILED_INFORMATION
+    SHIPPING_ADDRESS
+}
+
+type GiftRegistryRegistrant {
+    id: ID!
+    first_name: String!
+    last_name: String!
+    email: String! @doc(description: "Accessible to the registry owner only")
+    dynamic_attributes: [GiftRegistryRegistrantDynamicAttribute]
```

Comment:
Hi @paliarush, do you mind if I get some clarification here. What exactly `dynamic_attributes` are supposed to be for registrant? Or it's something that has to be extended later on?
Thank you.

Reply:
Please take a look at screenshots here or play with gift registry in the admin: https://docs.magento.com/user-guide/marketing/gift-registry-create.html

Gift registry has its own system of dynamic attributes, which is not EAV. 

Please let me know if this does not answer your question.

Reply:
I didn't pay attention to the `Attribute Group`.
Thank you.